File: KnightsTour.cpp
#include <iostream>
int board[8][8];
bool place(int row, int col, int val, int board[8][8])
{
    if(val == 64)
    {
        return true;
    }
    
    if(row<0 || col<0 || row>=8 || col>=8 || board[row][col] != -1)
    {
        return false;
    }
    //(2, -1), (2, 1), (1, 2), (-2, 1), (-1, 2), (-1, -2), (1, -2), (-2, -1)
    board[row][col] = val;
    if(place(row+2,col-1,val+1, board)){return true;}
    if(place(row+2,col+1,val+1, board)){return true;}
    if(place(row+1,col+2,val+1, board)){return true;}
    if(place(row-2,col+1,val+1, board)){return true;}
    if(place(row-1,col+2,val+1, board)){return true;}
    if(place(row-1,col-2,val+1, board)){return true;}
    if(place(row+1,col-2,val+1, board)){return true;}
    if(place(row-2,col-1,val+1, board)){return true;}
    board[row][col] = -1;
    return false;
    
}

void print(int param[8][8])
{
    for(int i = 0; i < 8; i++)
    {
        std::cout << "\n";
        for(int j = 0; j < 8; j++)
        {
            std::cout << " " << ((param[i][j]<10)?"0":"") << param[i][j];
        }
    }
}

int main() {
    // Write C++ code here
    std::cout << "Knight's Journey:";
    int i,j;
    for(i=0; i < 8; i++)
    {
        for(j=0; j < 8; j++)
        {
            board[i][j] = -1;
        }
    }
    place(0,0,0,board);
    print(board);
    return 0;
}

File: KnightsTour/KnightsTour.py

class Board:
    def __init__(self, rows, cols):
        self.board = [[-1 for i in range(rows)] for j in range(cols)]#fill board with negative ones
        self.order = [(2, -1), (2, 1), (1, 2), (-2, 1), (-1, 2), (-1, -2), (1, -2), (-2, -1)]
        self.rows = len(self.board)
        self.cols = len(self.board[0])

    def display(self):
        print("\n")
        for row in self.board:
            print("\n")
            for col in row:
                print((" " if col < 10 else "") + str(col), end = "  ")
        print("\n")

    def placeKnight(self,row,col,val):
        #return true if board is filled
        if(val == self.rows *self.cols):
            return True

        #if spot full >> return false
        if(row >= self.rows or row < 0 or col >= self.cols or col<0 or self.board[row][col]!= -1):
            return False
        
        #else >> place knight
        self.board[row][col] = val
        
        #check all possible next placements
        for spot in self.order:
            (newR,newC) = spot
            #if they work >> return true
            if(self.placeKnight(newR+row,newC+col,val+1)):
                return True

        #unplace Knight and return false
        self.board[row][col] = -1
        return False
    
    def go(self):
        self.placeKnight(0,0,0)
        self.display()
b=Board(8,8)
b.go()
